src/core/InterleavedBufferAttribute.js

```diff
@@ -134,6 +135,25 @@ Object.assign( InterleavedBufferAttribute.prototype, {
 
 		return this;
 
+	},
+
+	clone: function () {
+
+		var count = this.count;
+		var itemSize = this.itemSize;
+		var array = this.data.array.slice( 0, count * itemSize );
+
+		for ( var i = 0; i < count; ++ i ) {
+
+			array[ i ] = this.getX( i );
+			if ( itemSize >= 2 ) array[ i + 1 ] = this.getY( i );
+			if ( itemSize >= 3 ) array[ i + 2 ] = this.getZ( i );
+			if ( itemSize >= 4 ) array[ i + 3 ] = this.getW( i );
+
+		}
+
+		return new BufferAttribute( array, itemSize, this.normalized );
```

Comment:
I would expect an `InterleavedBufferAttribute` to be returned from the `clone()` method.

Reply:
That seems more intuitive, yes, but I can't think of a reason you'd want data the attribute is interleaved with to also be copied... do we return a new InterleavedBufferAttribute with `stride === itemSize`? Or rename this as `toBufferAttribute()`?

Reply:
Only guessing here, but maybe
```javascript
BufferAttribute.setFromInterleavedBufferAttribute()
```
is what you want.

Reply:
Do you mean a static method, or instance like `geometry.fromBufferGeometry(bufferGeometry)`? Feels non-ideal making an empty array just to instantiate an empty buffer attribute then overwrite it. What about `.toNonInterleaved()` similar to `.toNonIndexed()`?

Reply:
I'm not sure if the problem you are trying to solve warrants a special method... How often do uses want to de-interleave their geometry?

Reply:
> How often do uses want to de-interleave their geometry?

Not sure, I only need this within GLTFLoader. But the idea that `.clone()` should exist for what most users will assume is a subclass of BufferAttribute seems nice on least-surprise principle.  The original version of this PR put the method inside of GLTFLoader, and I changed that based on https://github.com/mrdoob/three.js/pull/12800#issuecomment-349231832.

Reply:
I think your suggestion makes sense.
```javascript
var bufferAttribute = interleavedBufferAttribute.toNonInterleaved();

var geometry = new THREE.BufferGeometry();
geometry.addAttribute( 'position', bufferAttribute );
```
I'm just not sure if such a method is needed outside your loader... I'll leave that up to you.

Reply:
@mrdoob with the context above, any preference on moving this back into a private `cloneBufferAttribute(attr)` helper, vs renaming `clone()` as `toNonInterleaved()` here?

I don't expect users outside GLTFLoader to call `.toNonInterleaved()` directly, but we have multiple open issues about `.clone()` not working consistently, and this is one of those edge cases — having any interleaved buffer attributes will cause `bufferGeometry.clone()` to throw a TypeError.

Reply:
Agreed. I think this is an accurate statement of the issues:

1. @donmccurdy wants to de-interleave his buffer geometry. I think that should be implemented via a private helper for now.

2. `BufferGeometry.clone()` does not support interleaved attributes. That should be fixed.

Reply:
I've looked into this and created a little code snippet that should enable the support of interleaved attributes in `BufferGeometry.clone()`. The most important thing first: Because all interleaved attributes of a geometry share a common instance of [ArrayBuffer](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer), it does not make sense to perform an isolated clone for a single interleaved attribute. A clone does only make sense on geometry level in order to ensure continuous, interleaved memory. The following code performs this operation. The assumption of the implementation is that all interleaved attributes of a geometry share the same array buffer.

```javascript
// attributes

var attributes = source.attributes;
var arrayBuffer;

for ( name in attributes ) {

	var attribute = attributes[ name ];

	if ( attribute.isInterleavedBufferAttribute ) {

		if ( arrayBuffer === undefined ) {

                        // clone the array buffer once and use it for subsequent typed arrays
			arrayBuffer = attribute.array.buffer.slice( 0 ); 

		}

                // now perform the actual clone of the attribute and its internal data
		var typedArray = new attribute.array.constructor( arrayBuffer );
		var interleavedBuffer = new attribute.data.constructor( typedArray, attribute.data.stride );
		var interleavedBufferAttribute = new attribute.constructor( interleavedBuffer, attribute.itemSize, attribute.offset, attribute.normalized );

		this.addAttribute( name, interleavedBufferAttribute );


	} else {

		this.addAttribute( name, attribute.clone() );

	}

}
```
I also suggest to remove `.copy()` and `.clone()` of `InterleavedBuffer` since these methods make actually no sense. `.copy()` creates a new array buffer for each instance of `InterleavedBuffer` which results in non-interleaved memory. Besides, it's maybe a good idea to merge `InterleavedBuffer` with `InterleavedBufferAttribute`. Properties like `dynamic` and `updateRange` should be part of the attribute. Are there any reason for a separation of `InterleavedBuffer` and `InterleavedBufferAttribute`?

Reply:
And yeah, the code is not nice but it illustrates the approach pretty well :grin: